statsd/telemetry.go

```diff
@@ -110,6 +110,18 @@ type Telemetry struct {
 	// AggregationNbContextTiming is the total number of contexts for timings flushed by the aggregator when either
 	// WithClientSideAggregation or WithExtendedClientSideAggregation options are enabled.
 	AggregationNbContextTiming uint64
+	// AggregationNbSample is the total number of samples flushed by the aggregator when either
+	// WithClientSideAggregation or WithExtendedClientSideAggregation options are enabled.
+	AggregationNbSample uint64
+	// AggregationNbSampleHistogram is the total number of samples for histograms flushed by the aggregator when either
+	// WithClientSideAggregation or WithExtendedClientSideAggregation options are enabled.
+	AggregationNbSampleHistogram uint64
+	// AggregationNbSampleDistribution is the total number of samples for distributions flushed by the aggregator when
+	// either WithClientSideAggregation or WithExtendedClientSideAggregation options are enabled.
+	AggregationNbSampleDistribution uint64
+	// AggregationNbSampleTiming is the total number of samples for timings flushed by the aggregator when either
+	// WithClientSideAggregation or WithExtendedClientSideAggregation options are enabled.
+	AggregationNbSampleTiming uint64
```

Comment:
Ah, sorry, would you like me to make a separate PR to update the telemetry? I can't really separate the telemetry with the rest of the logic since this is how we're tracking whether or not samples were dropped for the benchmarks.

Reply:
In all honesty, the best would be a separate PR for the telemetry and a separate one for the "rand" replacement.

The reasoning behind this is that they are the two things actively modifying the current behaviour of the library, while the lossy mode isn't really (it's fairly isolated because of the switch case).

If the telemetry can't be extracted to a different PR, it would still be preferable that the rand change were extracted to a separate PR.

Reply:
Understood!

I'll get rid of the rand changes for now and introduce them in a separate, later PR.